### PR TITLE
Tidy q3TiledAligner

### PR DESCRIPTION
**Pull request by holmeso**
# Description

Renamed the only class in this project from Q3TiledAligner to GenerateTiledAlignerFile, hopefully making the functions that the class performs more apparent.
This is to prepare the project for an up coming injection of code relating to the tiled aligner.

# How Has This Been Tested?

I have run the updated code against a fa file, and it has produced output that looks good.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java [hunk 1]
```diff
@@ -67,33 +75,29 @@ void performBinning() throws IOException {
 				break;
 			}
 			
-			chrLengthPosition.add(currentChr + ":" + referenceBasesLength + ":" + longPosition); 
+			chrLengthPosition.add(currentChr + Constants.COLON + referenceBasesLength + Constants.COLON + longPosition);
```
Comment (delocalizer) [inline q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java:78]: For my education only, what does `Constants.COLON` etc. give you that a literal ":" does not?

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java [hunk 2]
```diff
@@ -111,67 +115,110 @@ void performBinning() throws IOException {
...
 				sb.append("\n");
-				writer.write(sb.toString());
+				writer.write(sb.toString().getBytes());
 			}
 		}
+		
+		
+		
+		
+//
```
Comment (delocalizer) [inline q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java:179]: delete this block comment?

**Review by delocalizer — APPROVED** [review]

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java [hunk 3]
```diff
@@ -39,10 +48,9 @@
 	private byte currentChrIndex;
 	
 	private final int tileSize = 13;
-	private final int positionsCutoff = 499;
+	private final int positionsCutoff = 4999;
```
[the remark this answers is not in the document]
Comment (ChristinaXu2017) [reply q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java:51]: I agree, but not sure whether the config file exist or not

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java (hunk 1 above)
Comment (ChristinaXu2017) [reply q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java:78]: Constants.COLON = ":"

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java (hunk 2 above)
Comment (ChristinaXu2017) [reply q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java:179]: agree

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (holmeso) [reply q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java]: yep, have created an option for this

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java (hunk 1 above)
Comment (holmeso) [reply q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java:78]: I think this harks back to the days before Java would intern strings. Perhaps using a static variable removed the need for the lookup in the String pool...

File: q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java (hunk 2 above)
Comment (holmeso) [reply q3tiledaligner/src/au/edu/qimr/tiledaligner/GenerateTiledAlignerFile.java]: done

**Comment by holmeso** [comment]
Thanks for the feedback. I've pushed changes which unfortunately means you will need to re-approve....

**Review by delocalizer — APPROVED** [review]

**Pull request merged**